### DISCUSSION TITLE
solid-start Locator.js support

### DISCUSSION THREAD

**Discussion by nirtamir2** [thetarnav/solid-devtools#163] [Q&A]
Hi,
I try to config this extension to work with solid-start.
Added `devtools({ name: true })` as the first `vite.config.ts` plugin.
It seems like the browser extension work, and if I explicitly add `useLocator` at `entry-client.tsx` I can mark the components.
But LocatorJS is not able to find the component in the code when I clicked the marked component name.
Can we add an example of how to integrate it with LocatorJS?

Thanks!

**Comment by thetarnav** [comment]
Hey,
Are you referring to [LocatorJS](https://www.locatorjs.com) or a [similar feature of solid debugger](https://github.com/thetarnav/solid-devtools/tree/main/packages/debugger#using-locator)? (which was inspired by locatorjs, so I named it the same way)

It seems that the mention of LocatorJS was removed from the readme when I was moving it to the debugger package... that's not good—may be confusing.

**Reply by nirtamir2** [reply]
Ahhh
Yes, I misunderstood it because of the name LocatorJS.
So [locatorjs](https://www.locatorjs.com/) won't work with solid-start, but [solid-devtools](https://github.com/thetarnav/solid-devtools)'s locatorJS should work. OK
Thanks!

**Reply by nirtamir2** [reply]
BTW I notice that the locator feature of solid-devtools could mark the component, but not open it in the IDE.
I did some configuration in vite.config.ts and entry-client like in the solid example. What did I miss? Is it working?

**Reply by thetarnav** [reply]
Have you passed the `targetIDE` option to `useLocator`, and enabled `jsxLocation` in the vite plugin?
This is the only setup neccessary. For me it is always working, but there were situations when it had some troubles on linux/mac. They had been resolved though..

**Reply by nirtamir2** [reply]
Now after the updates I'm able to see the component name after pressing the `cmd` key and hovering over the component. When the chrome extension is open and I click on the component with the `cmd` key pressed, it opens the matching component on the chrome extension page. I guess the demo of opening the source file in the IDE is still not possible, right?

Because I see the https://github.com/thetarnav/solid-devtools/tree/main/packages/debugger#using-the-locator-on-the-page video, and I want to enable this feature somehow.

**Reply by nirtamir2** [reply]
My project is https://github.com/nirtamir2/solid-start-starter

**Reply by thetarnav** [reply]
When the extension is opened the click action will reveal it in the extension. If it's closed, the component location should open in the selected IDE.
There is a way to open component location from the extension as well:
https://github.com/thetarnav/solid-devtools/tree/main/packages/transform#locatorcomponentlocation

**Reply by thetarnav** [reply]
Btw the config changed a little bit recently, there is no need to import anything or call `useLocator` in the entry file anymore.
All can be done just using the vite plugin. (https://github.com/thetarnav/solid-devtools/tree/main/packages/extension#getting-started)

**Reply by nirtamir2** [reply]
Thanks!

When I only use the vite plugin the extension does not load/recognize solid devtools extension.
![image](https://user-images.githubusercontent.com/16452789/210096292-9d6df0d9-5e62-4ffb-b4b6-954916117947.png)

It seems that the example folder is outdated https://github.com/thetarnav/solid-devtools/tree/main/examples

Here is what I've tried:
```ts
import devtools from 'solid-devtools/vite'
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-expect-error
import vercel from "solid-start-vercel";
import solid from "solid-start/vite";
import { defineConfig } from "vite";
import solidSvg from "vite-plugin-solid-svg";

export default defineConfig(() => {
  return {
    plugins: [
      devtools({
        autoname: true,
        locator: {
          targetIDE: "webstorm",
          componentLocation: true,
          jsxLocation: true,
        },
      }),
      solid({
        adapter: vercel(),
      }),
      solidSvg({
        defaultAsComponent: true,
      }),
    ],
  };
});

```

**Reply by thetarnav** [reply]
hmmm
that's possibly a bug then...
And yes, most examples aren't updated yet. Gonna try to debug this and publish an update.

**Reply by nirtamir2** [reply]
Thanks for your help! I really appreciate it! ❤️

**Reply by thetarnav** [reply]
Just published an update. Should work better with solid start now.

**Reply by nirtamir2** [reply]
Great!
I can confirm that now the solid-devtools extension is detected out of the box with the vite plugin

![image](https://user-images.githubusercontent.com/16452789/210137672-8eb3b0ba-b029-4bcf-ad35-51503da403a3.png)

**Reply by nirtamir2** [reply]
Still, I'm not able to navigate to the component source code. I can only see which component is it in the devtools extension (if it is open).
